checkov/secrets/plugins/entropy_keyword_combinator.py

```diff
@@ -34,26 +108,151 @@ def analyze_line(
+                if keyword_on_key:
+                    return self.detect_secret(self.high_entropy_scanners_iac, filename, line, line_number, **kwargs)
+
+                # not so classic key-value pair, from multiline, that is only in an array format.
+                # The scan is one-way backwards, so no duplicates expected.
+                elif filetype == FileType.YAML:
+                    return self.analyze_iac_line_yml(
+                        filename=filename,
+                        line=line,
+                        line_number=line_number,
+                        context=context,
+                        raw_context=raw_context,
+                        value_pattern=value_keyword_regex_to_group,
+                        secret_pattern=secret_keyword_regex_to_group,
+                        kwargs=kwargs
+                    )
             else:
-                for entropy_scanner in self.high_entropy_scanners:
-                    matches = entropy_scanner.analyze_line(filename, line, line_number, **kwargs)
-                    if matches:
-                        return matches
+                return self.detect_secret(self.high_entropy_scanners, filename, line, line_number, **kwargs)
         return set()
 
-    def analyze_string(self, string: str) -> Generator[str, None, None]:
-        raise NotImplementedError()
+    def analyze_iac_line_yml(
+            self,
+            filename: str,
+            line: str,
+            line_number: int = 0,
+            context: CodeSnippet | None = None,
+            raw_context: CodeSnippet | None = None,
+            value_pattern: Optional[dict[Pattern[str], int]] = None,
+            secret_pattern: Optional[dict[Pattern[str], int]] = None,
+            **kwargs: Any,
+    ) -> set[PotentialSecret]:
+        secrets = set()
+        if context is not None and raw_context is not None:
+            i = context.target_index
+
+            value_secret = self.extract_from_string(pattern=secret_pattern, string=context.lines[i])
+            secret_adjust = self.format_reducing_noise_secret(value_secret)
+            entropy_on_value = self.detect_secret(self.high_entropy_scanners, filename, secret_adjust, line_number, **kwargs)
+
+            if entropy_on_value:
+                possible_keywords: set[str] = set()
+                forward_range = range(context.target_index - 1, -1, -1)
+                backwards_range = range(context.target_index + 1, len(context.lines))
+                possible_keywords |= self.get_lines_from_same_object(forward_range, context, raw_context)
+                possible_keywords |= self.get_lines_from_same_object(backwards_range, context, raw_context)
+
+                for other_value in possible_keywords:
+                    if self.extract_from_string(pattern=value_pattern, string=other_value):
+                        secrets |= entropy_on_value
+                    if secrets:
+                        break
```

Comment:
```suggestion
                        secrets |= entropy_on_value
                        break
```
should be the same or am I missing something?

Reply:
changed it all! :) thanks for the detailed review!